tests/mcproto/packets/handshaking/test_handshake.py

```diff
@@ -1,101 +1,38 @@
-            bytes.fromhex("f5050f6d632e61697263732e726163696e6763dd01"),
-        ),
-        (
-            {
-                "protocol_version": 757,
-                "server_address": "hypixel.net",
-                "server_port": 25565,
-                "next_state": NextState.LOGIN,
-            },
-            bytes.fromhex("f5050b6879706978656c2e6e657463dd02"),
-        ),
-        (
-            {
-                "protocol_version": 757,
-                "server_address": "hypixel.net",
-                "server_port": 25565,
-                "next_state": NextState.STATUS,
-            },
-            bytes.fromhex("f5050b6879706978656c2e6e657463dd01"),
-        ),
+from tests.helpers import gen_serializable_test
+
+gen_serializable_test(
+    context=globals(),
+    cls=Handshake,
+    fields=[
+        ("protocol_version", int),
+        ("server_address", str),
+        ("server_port", int),
+        ("next_state", NextState),
     ],
-)
-def test_serialize(kwargs: dict[str, Any], expected_bytes: list[int]):
-    """Test serialization of Handshake packet."""
-    handshake = Handshake(**kwargs)
-    assert handshake.serialize().flush() == bytearray(expected_bytes)
-
-
-@pytest.mark.parametrize(
-    ("read_bytes", "expected_out"),
-    [
+    test_data=[
         (
+            (757, "mc.aircs.racing", 25565, NextState.LOGIN),
             bytes.fromhex("f5050f6d632e61697263732e726163696e6763dd02"),
-            {
-                "protocol_version": 757,
-                "server_address": "mc.aircs.racing",
-                "server_port": 25565,
-                "next_state": NextState.LOGIN,
-            },
         ),
         (
+            (757, "mc.aircs.racing", 25565, NextState.STATUS),
             bytes.fromhex("f5050f6d632e61697263732e726163696e6763dd01"),
-            {
-                "protocol_version": 757,
-                "server_address": "mc.aircs.racing",
-                "server_port": 25565,
-                "next_state": NextState.STATUS,
-            },
         ),
         (
+            (757, "hypixel.net", 25565, NextState.LOGIN),
             bytes.fromhex("f5050b6879706978656c2e6e657463dd02"),
-            {
-                "protocol_version": 757,
-                "server_address": "hypixel.net",
-                "server_port": 25565,
-                "next_state": NextState.LOGIN,
-            },
         ),
         (
+            (757, "hypixel.net", 25565, NextState.STATUS),
             bytes.fromhex("f5050b6879706978656c2e6e657463dd01"),
-            {
-                "protocol_version": 757,
-                "server_address": "hypixel.net",
-                "server_port": 25565,
-                "next_state": NextState.STATUS,
-            },
         ),
+        # Invalid next state
+        ((757, "localhost", 25565, 3), ValueError),
+        ((757, "localhost", 25565, 4), ValueError),
+        ((757, "localhost", 25565, 5), ValueError),
+        ((757, "localhost", 25565, 6), ValueError),
```

Comment:
Why so many? 2 is enough, maybe even just 1 for the `next_state=3`.

Reply:
Yeah, idk what happened